### ISSUE TITLE
Add operations-based status information as output

### ISSUE THREAD

**Issue by nicholashagen** [teatrove/teatrove#196]
Add a status template that provides status information from configurable MBeans and other sources that provides the health of the application.  This can then be hooked into other monitoring appliances.

**Comment by philipseidel** [comment]
This is an excellent idea.  Here are a few thoughts on where I would like to see something like this go.
- Standard interface with which to interact.  The protocol doesn't necessarily need to be the same (ie. http, jmx, etc.) but we should expect that the interaction between multiple interfaces will return the same results.
- It needs to have a standard set of metrics that are collected and exposed out of the gate with no work on engineering's part aside from including the library and exposing the interface.
- It needs to be easily extensible so that engineering can push metrics to it with ease and without the need to coordinate with operations.
- It needs to allow for the application to send signals (ie. I am currently in error because of...) that something is wrong so that monitoring tools don't have to perform multiple types of queries to figure out if the service is healthy or not.  This is one of the most important pieces in my opinion as monitoring systems can't monitor every endpoint of an application to see if it is healthy.
- It should be easily parsed and human readable so that we can look at this at any time to get a general idea of what is going on in the service.
  - How many database connections does it have open and to which server?
  - Timing statements for long running queries
  - Metrics on important templates, functions, etc.
  - Basically, anything that can be instrumented should be exposed easily

The goal should be to have the service report its health based on the role of the application as operations is not close enough to the application to make these decisions.  The engineer that writes the application knows what is and isn't important so if we can get them to make some decisions and a way to easily expose these data points it will take monitoring to the next level.
